src/ucs/arch/x86_64/cpu.c

```diff
@@ -707,6 +732,317 @@ ucs_status_t ucs_arch_get_cache_size(size_t *cache_sizes)
     return cache_count == UCS_CPU_CACHE_LAST ? UCS_OK : UCS_ERR_UNSUPPORTED;
 }
 
+#if __AVX__
+static UCS_F_ALWAYS_INLINE
+size_t ucs_x86_nt_dst_buffer_transfer(void *dst, const void *src, size_t len,
```

Comment:
given the functions, it could make sense to add simple copying tests (iterate over src aligment x iterate over length x iterate over dst alignment, for short mid and big lengts)

Reply:
@tvegas1 Can you please elaborate? I did not understand this, please provide some examples as well? (may be tests done for builtin-memcpy?)

Reply:
since `ucs_x86_nt_buffer_transfer()`/`ucs_x86_nt_dst_buffer_transfer()`/`ucs_x86_nt_src_buffer_transfer()` are memcpy functions now implemented in ucs, i was wondering if we could test that they are properly copying memory, possibly under `test/gtest`, with something like:
```C
<src/dest big enough 256B aligned>
for (len = 0; len < 8*1024; len++) {
  for (i = 0; i < 256; i++) {
    for (j = 0; j < 256; j++) {
      <memset pattern in src>
      ucs_x86_nt_dst_buffer_transfer(dest + i, src + i, len);
      <memcmp pattern in dst>
    }
  }
}

```

Reply:
Done. Thanks for the suggestion.

Reply:
@tvegas1 most of the builds are failing now with the below error due to this gtest addition. Please see the logs below.


/__w/1/s/contrib/../test/gtest/ucs/arch/test_x86_64.cc:98: undefined reference to `ucs_x86_nt_buffer_transfer'
collect2: error: ld returned 1 exit status
make[3]: *** [Makefile:1429: gtest] Error 1
make[3]: Leaving directory '/home/swx-azure-svc_azpcontainer/80290-20240430.3/build/test/gtest'
make[2]: *** [Makefile:3425: all-recursive] Error 1
make[2]: Leaving directory '/home/swx-azure-svc_azpcontainer/80290-20240430.3/build/test/gtest'
make[1]: *** [Makefile:790: all-recursive] Error 1
make[1]: Leaving directory '/home/swx-azure-svc_azpcontainer/80290-20240430.3/build'
make: *** [Makefile:654: all] Error 2

Could you please help to identify the problem?

Reply:
i think the build might be missing define `__AVX__`

Reply:
@tvegas1 Thanks for catching it, I was always building with -mavx flag locally, so this issue never came there.